### DISCUSSION TITLE
Suppressions documentation

### DISCUSSION THREAD

**Discussion by jshapiro2** [postalserver/postal#2363] [Help with using Postal]
Hello All,

I unable to locate any documentation on how suppressions work?  At what point is an email moved to the suppressed list?

**Comment by slobodangacesa** [comment]
After two hard fails, address it put into suppression list for 30 days.
You can always try re-delivering message that failed or was held for that address, and if delivery was successful, address will be removed from supression list.

**Comment by jshapiro2** [comment]
Are there any configurations on how often it retries an email before moving to suppressions?
For example, if there is no smtp or no mx record, why would you want it to keep trying on those?

**Reply by willpower232** [reply]
Unfortunately it is not straightforward to fully tell the difference between different kinds of errors as the error messages will not be consistent between providers. It is also impossible to tell the difference between a temporary problem with DNS or the network and a typo so Postal errs on the side of temporary errors to give the messages a proper chance.

You can see many things are classed as softfail here which means they'll be tried up to 18 times by default, with increasing amounts of time between each attempt https://github.com/postalserver/postal/blob/main/lib/postal/smtp_sender.rb

The number of attempts can be changed in the config file https://github.com/postalserver/postal/blob/c42dd1b9f6c55a75dcaea74a725e5f8703729ec7/config/postal.defaults.yml#L14
